### PR TITLE
CI: add dependabot.yml to auto-update dependencies

### PR DESCRIPTION
**Pull request by aciba90**
GH actions and lint requirements will be try to be updated in autogenerated PRs.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 2
+updates:
+  - package-ecosystem: "pip"
```
Comment (aciba90) [inline .github/dependabot.yml:3]: This part will basically mimic the tip targets in Jenkins by creating a PR with updated dependencies and the added benefit of:

2. If the build passes, we can straight merge those PRs.
3. If it does not, we see that some style needs to be fixed.

If we try and like this approach, then we could remove the tip jobs in Jenkins.

Comment (aciba90) [reply .github/dependabot.yml:3]: This is false, because tip targets check against the last version of python.

Comment (paride) [reply .github/dependabot.yml:3]: I like this approach, but I can't find documentation on which requirement files dependabot will update, at least not in [1]. Will it update `dev/requirements.txt` and `test-requirements.txt`? Do you happen to have a reference for this?

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

Comment (aciba90) [reply .github/dependabot.yml:3]: AFAICT, at the moment one only can specify the directory to search the manifest files and `dependabot` will discover them there. That's why I put the file under `dev/` and configure it in line 4.

I found some discussions explaining that behavior as for example: https://github.com/dependabot/dependabot-core/issues/3940#issuecomment-1227005267

The source code is here: https://github.com/dependabot/dependabot-core/blob/main/python/helpers/lib/parser.py

Comment (paride) [reply .github/dependabot.yml:3]: Found it, apparently it probes `**/.txt`:
```
    requirement_files = glob.glob(os.path.join(directory, '*.txt')) \
        + glob.glob(os.path.join(directory, '**', '*.txt'))
```
Thanks for digging it up!

**Review by paride — APPROVED** [review]
LGTM, thanks!

**Comment by paride** [comment]
@aciba90 is this actually still a draft?

**Comment by aciba90** [comment]
> @aciba90 is this actually still a draft?

Ready! Thanks, @paride for the review. I wanted to make sure that the default configuration does not auto-merge PRs or anything weird. Checked and it looks safe: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/managing-pull-requests-for-dependency-updates

I am going to ping other as reviewers.

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**